tests/Silex/Tests/Provider/TwigServiceProviderTest.php

```diff
@@ -26,6 +26,7 @@ class TwigServiceProviderTest extends \PHPUnit_Framework_TestCase
     public function testRegisterAndRender()
     {
         $app = new Application();
+        $this->setMockFormFactoryForApplication($app);
```

Comment:
I think you should have `$app = $this->addFormFactoryMockToApplication($app)`. Do you really need a mock of the FormFactory here? Looks like enough when the application has the key `form.factory`, there is no additional type check in the part you want to test.

Reply:
So far mock of the FormFactory is not required here. I thought it was not safe enough to set is as an empty callback, the code inside if statement could rely on the correct value of `form.factory`.

Reply:
Is my guess about callback correct? Do you think it should be like this?
`$app['form.factory'] = function () {
     return true;
 }`

Is it required change?

Reply:
You could even use `$app['form.factory'] = true`